Public/Src/Cache/ContentStore/Distributed/Utilities/GrpcDistributedPathTransformer.cs

```diff
@@ -16,15 +18,22 @@ namespace BuildXL.Cache.ContentStore.Distributed.Utilities
     /// </summary>
     public class GrpcDistributedPathTransformer : IAbsolutePathTransformer
     {
+        private readonly IReadOnlyDictionary<string, string> _junctionsByDirectory;
         private static readonly string _localMachineName = Environment.MachineName;
         internal const string BlobFileExtension = ".blob";
 
+        /// <nodoc />
+        public GrpcDistributedPathTransformer(IReadOnlyDictionary<string, string> junctionsByDirectory = null)
+        {
+            _junctionsByDirectory = junctionsByDirectory ?? new Dictionary<string,string>();

```

Comment:
Does the dictionary need a comparer based on OS? I.e. case insensitve on windows, case sensitive for linux and dynamic for mac?

Reply:
See below that every path is just ToUppered anyway... perhaps a comment to the effect would be nice.
side-question: Do we have any assumptions on the network share supporting path transformations? If we have a unix box with an Samba server I recall that the system was mixed case sensitive depending on how you use it...

---
In reply to: [275859026](https://github.com/Microsoft/BuildXL/pull/148#discussion_r275859026) [](ancestors = 275859026)